massa-execution-worker/src/context.rs

```diff
@@ -145,9 +145,26 @@ impl ExecutionContext {
         }
     }
 
-    /// Resets context to an existing snapshot
+    /// Resets context to an existing snapshot.
+    /// Optionally emits an error as an event after restoring the snapshot.
     /// Note that the snapshot does not include slot-level information such as the slot number or block ID.
-    pub fn reset_to_snapshot(&mut self, snapshot: ExecutionContextSnapshot) {
+    ///
+    /// # Arguments
+    /// * `snapshot`: a saved snapshot to be restored
+    /// * `with_error`: an optional execution error to emit as an event conserved after snapshot reset.
+    pub fn reset_to_snapshot(
+        &mut self,
+        snapshot: ExecutionContextSnapshot,
+        with_error: Option<ExecutionError>,
```

Comment:
I think that's because for now we use the reset_to_snapshot only in error contexts. We might setup some more advanced uses in the future (eg. context push/pop, try/catch etc...), which is why I left the option.